plugins/base/src/main/kotlin/renderers/html/htmlPreprocessors.kt

```diff
@@ -26,15 +26,21 @@ abstract class NavigationDataProvider {
             children = page.navigableChildren()
         )
 
-    private fun ContentPage.navigableChildren(): List<NavigationNode> =
-        when {
-            this !is ClasslikePageNode ->
-                children.filterIsInstance<ContentPage>().map { visit(it) }
-            documentables.any { it is DEnum } ->
-                children.filter { it is WithDocumentables && it.documentables.any { it is DEnumEntry } }
-                    .map { visit(it as ContentPage) }
-            else -> emptyList()
-        }.sortedBy { it.name.toLowerCase() }
+    private fun ContentPage.navigableChildren(): List<NavigationNode> {
+        return if (this !is ClasslikePageNode) {
+            children
+                .filterIsInstance<ContentPage>()
+                .map { visit(it) }
+                .sortedBy { it.name.toLowerCase() }
+        } else if (documentables.any { it is DEnum }) {
+            // no sorting for enum entries, should be the same as in source code
+            children
+                .filter { child -> child is WithDocumentables && child.documentables.any { it is DEnumEntry } }
+                .map { visit(it as ContentPage) }
```

Comment:
apparently we don't have any unit tests for navigation, only very basic, half-fake integration tests which are difficult to extend

I wonder if it's at all possible to write unit tests for navigation sidemenu, given how it's generated?